File: frontend/src/app/shared/models/league.ts
export class League {
    regionId: number = 1;
    socialMediaId: number = 1;
    name: string = "League";
    imagePath: string | null = '';
    description: string = '';
    colorHex: string = '#000'
    leagueUsers: any[] = [];
    region: any = null;
    seasons: any[] = [];
    socialMedia: any = null;
    id: number = 1;
}

File: RacingLeagueHub/BLL/Entities/League.cs
using RacingLeagueHub.BLL.Models.Enums;
using RacingLeagueHub.BLL.Entities.Seasons;

namespace RacingLeagueHub.BLL.Entities;

public class League : EntityBase
{
    public Region Region { get; set; }
    public string Name { get; set; }
    public string? Description { get; set; }
    public string Timezone { get; set; }
    public string Slug { get; set; }
    public long? LogoResourceId { get; set; }

    public virtual Resource? LogoResource { get; set; }

    public virtual ICollection<LeagueUser> LeagueUsers { get; set; }
    public virtual ICollection<Season> Seasons { get; set; }
}